crates/jit-icache-coherence/src/libc.rs

```diff
@@ -139,17 +139,44 @@ fn riscv_flush_icache(start: u64, end: u64) -> Result<()> {
     }
 }
 
+#[cfg(target_arch = "aarch64")]
+fn aarch64_flush_icache(start: u64, end: u64) {
+    use core::arch::asm;
+
+    // See `sys_icache_invalidate` implementation in Darwin at
+    // https://github.com/apple/darwin-libplatform/blob/main/src/cachecontrol/arm64/cache.s. It
+    // turns out that all of these instructions work in userspace, so
+    // we can do this portably without having to rely on OS-specific
+    // syscalls like `sys_cache_invalidate()` on macOS or something
+    // equivalent on Linux.
+    const CACHE_LINE_SIZE: u64 = 64;
+    // For each cache line, flush the icache.
+    let mut start = start & !(CACHE_LINE_SIZE - 1); // Round down.
+    let end = (end + CACHE_LINE_SIZE - 1) & !(CACHE_LINE_SIZE - 1); // Round up.
```

Comment:
Better to use https://doc.rust-lang.org/nightly/std/primitive.usize.html#method.next_multiple_of than to open-code this stuff so that readers have to double-check their hacker's delight knowledge.

```rust
let mut start = (start - 1).next_multiple_of(CACHE_LINE_SIZE);
let end = end.next_multiple_of(CACHE_LINE_SIZE);
```

Reply:
(FWIW, I verified that LLVM can recreate the hacker's delight style solution for calls to these methods with constant powers-of-two multiples.)